### ISSUE TITLE
asm2wasm fails on asm2wasm.h

### ISSUE THREAD

**Issue by bobvanluijt** [WebAssembly/binaryen#588]
Hi all,

I did the following on a simple hello world C script.
- `EMCC_WASM_BACKEND=1 emcc test.c -o test.js -s BINARYEN=1`

Next I wanted to use the asm2wasm script
- `./bin/asm2wasm test.js`

But it outputs:
`asm2wasm: /FOOBAR/binaryen-master/src/asm2wasm.h:113: char* wasm::Asm2WasmPreProcessor::process(char*): Assertion 'strstr(growthFuncStart + 1, "function ") == 0' failed.
Aborted (core dumped)`

Not sure what to do, any input is welcome (-:

**Comment by kripken** [comment]
test.js contains the JS part of the output. You should see the wasm part in `test.wasm`, etc. So asm2wasm can't work on that, it needs asm.js.

There wouldn't be asm.js in this case anyhow, since the wasm backend doesn't emit any. If you build using emscripten's llvm and asm.js backend, with the binaryen flag, then you would get `test.asm.js`.

**Comment by bobvanluijt** [comment]
Thanks @kripken 

You mention that I would get a `test.asm.js` if I added the binaryen flag, but `-s BINARYEN=1` is that flag, right? Or else, how would I add the binaryen flag to the emcc command?

**Comment by kripken** [comment]
Yes, `-s BINARYEN=1`. Same command basically, just without the wasm backend env var.

**Comment by bobvanluijt** [comment]
Aha, and this should be the emcc from the incoming branch, right?

**Comment by kripken** [comment]
Yes, older probably won't work.

**Comment by bobvanluijt** [comment]
Awesome, one last question. I don't seem to be able to find 3.9.0, the latest available is 3.8.0 http://llvm.org/releases/

**Comment by kripken** [comment]
The asm.js backend isn't in upstream llvm, you need to use emscripten's "fastcomp" fork.

**Comment by kripken** [comment]
(Can just use the emscripten SDK and it gets it all for you, or you can build it manually too.)

**Comment by tlively** [comment]
asm2wasm no longer exists!
